aten/src/ATen/native/native_functions.yaml

```diff
@@ -462,8 +462,10 @@
   device_check: NoCheck   # TensorIterator
   structured: True
   structured_inherits: TensorIteratorBase
+  ufunc_inner_loop:
+    Generic: add (AllAndComplex, BFloat16, Half)
```

Comment:
how would different supported dtypes on cuda and cpu be handled?

Reply:
This wouldn't be possible as is. I could introduce CPUGeneric/CUDAGeneric keys for those cases though. Do we have any concrete examples?

Reply:
e.g. `abs` supports `bool` on cuda but not on cpu https://fburl.com/code/8ehpvx2r (why is bool supported on cuda?)
`acos` (and many trig functions) support half on cuda but not on cpu https://fburl.com/code/28c9s51h (not supporting half on cpu is reasonable, it's dead slow anyway, no one should use it), etc. If `dtypesIfCUDA` is specified in OpInfos that means there's a discrepancy in datatypes support.

Reply:
Alright, CPUGeneric and CUDAGeneric it is, then.

Reply:
I'll do this in a follow up, issue filed here https://github.com/pytorch/pytorch/issues/73425